hapi/hedera-protobuf-java-api/src/main/proto/platform/state/virtual_map_state.proto

```diff
@@ -54,6 +45,42 @@ import "services/state/tss/tss_encryption_keys.proto";
 import "services/state/tss/tss_message_map_key.proto";
 import "services/state/tss/tss_vote_map_key.proto";
 
+/**
+ * This message defines the Virtual Leaf.
+ */
+message VirtualLeaf {
+  /**
+   * Virtual Key.<br/>
+   * The key used to identify and locate this virtual map entry.
+   */
+  VirtualMapKey key = 2;
+
+  /**
+   * Virtual Value.<br/>
+   * The actual data content stored for this virtual map entry.
+   */
+  VirtualMapValue value = 3;
+
+}
+
+/**
+ * This message defines the queue state used in the Virtual Map State.
```

Comment:
"State" is, unfortunately, too overloaded. "Queue state" here is not a state of type Queue (versus Singleton or KV), but an internal state of a queue state. Please, add some words about this. This state contains two pointers (longs), which indicate at what position a new element is put to the queue (tail) or removed from the queue (head). These aren't indices in any array or whatsoever, they are just numbers, which are increased on every operation, used to maintain elements order in the queue